docs/concepts/writing-config.md

```diff
@@ -0,0 +1,433 @@
+message Metric {
+        string descriptor_name = 1;
+        string value = 2;
+        map<string, string> labels = 3;
+}
+
+message MetricsParams {
+    repeated Metric metrics = 1;
+}
+</pre>
+    </td>
+    <td>
+<pre>
+{
+  "metrics": [
+    {
+      "descriptorName": "request_count",
+      "value": "1",
+      "labels": {
+            "source": "origin.ip",
+            "target": "target.service",
+      },
+    }, {
+      "descriptorName": "request_latency",
+      "value": "response.duration",
+      "labels": {
+            "source": "origin.ip",
+            "target": "target.service",
+      },
+    }
+  ]
+}  
+</pre>
+    </td>
+  </tr>
+</tbody>
+</table>
+
+*Note that we never name the outer-most message (the `MetricsParams`), but we do need the outer set of braces marking our JSON as an object. Also note that we shift from snake_case to lowerCamelCase; both are accepted but lowerCamelCase is idiomatic.*
+
+### Proto `enum` fields uses the name of the value:
+
+<table>
+  <tbody>
+    <tr>
+      <th>Proto</th>
+      <th>JSON</th>
+    </tr>
+    <tr>
+      <td>
+<pre>
+enum ValueType {
+    STRING = 1;
+    INT64 = 2;
+    DOUBLE = 3;
+    // more values omitted
+}
+
+message AttributeDescriptor {
+    string name = 1;
+    string description = 2;
+    ValueType value_type = 3;
+}
+</pre>
+      </td>
+      <td>
+<pre>
+{
+  "name": "request.duration",
+  "valueType": "INT64",
+}
+</pre>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+*Notice we don't provide a `description`: fields with no value can be omitted.*
+
+### Proto `message` fields can contain other messages:
+<table>
+  <tbody>
+    <tr>
+      <th>Proto</th>
+      <th>JSON</th>
+    </tr>
+    <tr>
+      <td>
+<pre>
+enum ValueType {
+    STRING = 1;
+    INT64 = 2;
+    DOUBLE = 3;
+    // more values omitted
+}
+
+message LabelDescriptor {
+    string name = 1;
+    string description = 2;
+    ValueType value_type = 3;
+}
+
+message MonitoredResourceDescriptor {
+  string name = 1;
+  string description = 2;
+  repeated LabelDescriptor labels = 3;
+}
+</pre>
+      </td>
+      <td>
+<pre>
+{
+  "name": "My Monitored Resource",
+  "labels": [
+    {
+      "name": "label one",
+      "valueType": "STRING",
+    }, {
+      "name": "second label",
+      "valueType": "DOUBLE",
+    },
+  ],
+}
+</pre>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+## Translating to YAML
+
+There is no canonical mapping between protobufs and YAML; instead protobuf defines a canonical mapping to JSON, and YAML defines a canonical mapping to JSON. To ingest YAML as a proto we convert it to JSON then to  protobuf.
+
+**Important things to note:**
+- YAML fields are implicitly strings
+- JSON arrays map to YAML lists; each element in a list is prefixed by a dash (`-`)
+- JSON objects map to a set of field names all at the same indentation level in YAML
+- YAML is whitespace sensitive and must use spaces; tabs are never allowed
+
+### Proto `map` and `message` fields:
```

Comment:
I thought it'd be more helpful to have everything side-by-side to make clear both the mapping from proto to yaml but also json to yaml. I can remove the JSON here but I thought it added value.